### PR TITLE
refactor update `blas/ext/base/dsortsh` to follow current project conventions

### LINKED ISSUE #1500: [RFC]: refactor `blas/ext/base/dsortsh` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dsortsh`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dsortsh) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by liberocks**
Resolves https://github.com/stdlib-js/stdlib/issues/1500.

## Description

> What is the purpose of this pull request?

This pull request refactors the existing implementation to follow the current project convention for `blas/ext/base/dsortsh`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves https://github.com/stdlib-js/stdlib/issues/1500
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dsortsh/benchmark/benchmark.mostly_sorted_few_uniques.js [hunk 1]
```diff
@@ -114,7 +107,7 @@ function main() {
 			'iterations': iter
 		};
 		bench( pkg+'::mostly_sorted,few_uniques:len='+len, opts, f );
-		iter = floor( pow( iter, 3.0/4.0 ) );
+		iter = pow( iter, 3.0/4.0 ) | 0;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsortsh/benchmark/benchmark.mostly_sorted_few_uniques.js]: I think we do not need this

File: lib/node_modules/@stdlib/blas/ext/base/dsortsh/benchmark/benchmark.mostly_sorted_few_uniques.native.js [hunk 2]
```diff
@@ -38,6 +38,7 @@ var dsortsh = tryRequire( resolve( __dirname, './../lib/dsortsh.native.js' ) );
 var opts = {
 	'skip': ( dsortsh instanceof Error )
 };
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsortsh/benchmark/benchmark.mostly_sorted_few_uniques.native.js]:
```suggestion
var rand = uniform( 1.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsortsh/benchmark/benchmark.mostly_sorted_few_uniques.native.js [hunk 3]
```diff
@@ -124,7 +113,7 @@ function main() {
 			'iterations': iter
 		};
 		bench( pkg+'::native,mostly_sorted,few_uniques:len='+len, bopts, f );
-		iter = floor( pow( iter, 3.0/4.0 ) );
+		iter = pow( iter, 3.0/4.0 ) | 0;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsortsh/benchmark/benchmark.mostly_sorted_few_uniques.native.js]: same comment here

Same comment follows on each of `benchmark.*`

File: lib/node_modules/@stdlib/blas/ext/base/dsortsh/manifest.json [hunk 4]
```diff
@@ -1,43 +1,47 @@
...
+      ],
+      "include": [
+        "./include"
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": [
+				"@stdlib/napi/export",
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsortsh/manifest.json]: Incorrect indentation

File: lib/node_modules/@stdlib/blas/ext/base/dsortsh/manifest.json [hunk 5]
```diff
@@ -1,43 +1,47 @@
...
+        "@stdlib/math/base/assert/is-nan",
+        "@stdlib/math/base/assert/is-negative-zero",
+				"@stdlib/napi/argv",
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsortsh/manifest.json]: same here

File: lib/node_modules/@stdlib/blas/ext/base/dsortsh/src/addon.c [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+	STDLIB_NAPI_ARGV_INT64( env, order, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, order, argv, 2 );
+
+	c_dsortsh( N, order, X, strideX );
+
+	return NULL;
+}
+
+STDLIB_NAPI_MODULE_EXPORT_FCN( addon )
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsortsh/src/addon.c]: add a trailing end line

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Left comments above, this PR is missing updates in readme, docs, etc. Please refer a similar PR and update accordingly. Thanks for your work!

**Comment by liberocks** [comment]
Hi @Pranavchiku I have made changes as suggested. However, it turns out we still need this code,
```
iter = pow( iter, 3.0/4.0 ) | 0;
```
I tried removing it and the benchmark goes forever. It causes timeout on the CI pipeline. Thus I put it back.

Anyway, thank you for reviewing my PR!

**Review by Planeshifter — APPROVED** [review]
Did some minor clean-up and will merge once CI passes. Thanks for your PR!

**Pull request merged**